docs/cpp_docs.rst

```diff
@@ -26,94 +44,556 @@ The beginning of both header and source code files should include
+    *             if `data` contains fewer than 2 elements
+    * @param[in] data the data to analyze
+    */
+   std::ostream operator<<(double & mean, double & stdDev, std::vector<double> const & data);
+
+When two or more consecutive parameters have *exactly* the same description, they can be combined:
+
+.. code-block:: cpp
+
+   /**
+    * @param x,y the coordinates where the function is evaluated
+    */
+
+Do not use a space between the parameters.
+
+.. _cpp-doxygen-parameters-inline:
+
+Annotating Parameters with Inline Comments (optional)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the parameter descriptions are very short, you may choose to annotate parameters with inline comments after each parameter, one per line.
+These comments are prefixed with ``///<``.
+
+For example:
+
+.. code-block:: cpp
+
+   /**
+    * Set a component's verbosity.
     *
+    * If no verbosity is specified, inherit from parent
     */
+   void setVerbosity(std::string const & name, ///< component of interest
+                     int verbosity); ///< desired trace verbosity
 
-.. _doc-cpp-package-definition:
+If the parameter descriptions are too long to fit in a single line of source, the ``@param`` documentation method must be used.
 
-Package Documentation / Definition
-==================================
+.. _cpp-doxygen-returns:
+
+Returns
+-------
+
+A ``@returns`` tag, followed by a description similar to the one for :ref:`cpp-doxygen-parameters`. If the returned value is a map, ensure that the key-value pairs are documented in the description.
+
+For consistency with Python documentation, always use ``@returns`` and not the synonymous ``@return``.
+
+.. _cpp-doxygen-throws:
+
+Throws
+-------
+
+A series of ``@throws`` tags, one for each type of exception (see :ref:`the style guide <style-guide-cpp-5-36>`). Each tag should have a description following the exception type. If the description extends over multiple lines, each line after the first must be indented.
+
+.. code-block:: cpp
+
+   /**
+    * Write an image to disk.
+    *
+    * @throws IoError Thrown if `fileName` could not be written to.
+    */
+   void writeImage(std::string const & fileName);
+
+Doxygen will render one or more ``@throws`` tags as a table of exceptions and descriptions, so do not treat ``@throws`` as the first word of the description.
+
+For consistency with Python documentation, always use ``@throws`` and not the synonymous ``@throw`` or ``@exception``.
+
+.. _cpp-doxygen-exceptsafe:
+
+Exception Safety
+----------------
+
+Whether or not there are any ``@throws`` tags, a function or method must have an ``@exceptsafe`` tag. (This applies even if the function or method is declared ``noexcept``, because the ``noexcept`` modifier is easy to miss in the generated documentation.) The description following the tag should describe the level of exception safety provided by the function or method.
```

Comment:
I was wondering why nobody objected to this bit at the RFC stage; I expected it to be the most controversial part of the proposal (granted, it was only one line in a page-long document).

I agree that much of the existing code is almost certainly not exception-safe. My hope was that requiring this in the documentation would encourage developers to pay attention to exception safety when writing new code, whereas making it optional would cause the issue to get quietly ignored. I took it for granted that nobody would be going through and systematically rewriting old documentation the way you describe.